src/wovnio/modified_vendor/SimpleHtmlDomNode.php

```diff
@@ -404,15 +404,8 @@ function replaceAttributeValue($name, $value) {
 
     function __isset($name)
     {
-        switch ($name)
-        {
-            case 'outertext': return true;
-            case 'innertext': return true;
-            case 'plaintext': return true;
-        }
-        //no value attr: nowrap, checked selected...
-        // return (array_key_exists($name, $this->attr)) ? true : isset($this->attr[$name]);
-        return false;
+        $val = $this->get_attr_val($this->attribute, $name);
+        return isset($val) && $val !== false ;
```

Comment:
Should false be considered not set?

Reply:
`isset(false)` is `true`.

Reply:
When attribute exists, $val will be value of attribute or `true`.
When attribute doesn't exist, $val will be `false`.

Reply:
Ah! I can replace by `return $val !== false ;`

Reply:
I see. I'm just a little confused about it because in JavaScript the `hasAtrribute()` function returns true even if the value of that attribute is false. I guess the use case in WOVN.php is slightly different in that we want to test if an attribute is explicitly defined as "valid".

![image](https://user-images.githubusercontent.com/11744696/98326880-9e87fd80-2035-11eb-921a-75e781fa9160.png)

Reply:
I think hasAttribute in WOVN.php will behave as same way as JavaScript.
You can see test cases in `testHasAttribute`.

Reply:
Ah I see. get_attr_val() will return a string "true" or "false". So it indeed behaves the same as JavaScript.